app/models/user.rb

```diff
@@ -110,4 +113,19 @@ def complete_registration
       registration.organization.update_attribute :active, true if registration.organization.present?
     end
   end
+
+  def technical_user_is_unique_in_org
+    # This validation make the assumption a technical user has only and only one membership at the creation
+    if User.joins(:organizations).where(name: name, 'organizations.id': memberships.first.organization.id).count > 0
+      errors.add :name, :must_be_unique_by_org
+    end
+  end
+
+  def normalize_name
+    self.normalized_name = if self.human?
```

Comment:
The second `self` is superfluous. (The first is needed because otherwise it would be a variable assignment.)